client/src/components/SearchReaslt/index.js

```diff
@@ -64,10 +65,19 @@ class SearchReaslt extends Component {
   };
 
   getData = () => {
-    const { history } = this.props;
-    const keyword = history.location.search.split('?')[1];
+    const { search } = this.props.location;
+    const values = queryString.parse(search);
     const array = [];
-    fetch(`/api/v1/search?q=${keyword}`)
+    let url = '';
+    if (values.category) {
+      const { category, exform, exto, incfrom, incto } = values;
+      url = `/api/v1/filter?incfrom=${incfrom}&incto=${incto}&exform=${exform}&exto=${exto}&category=${category}`;
```

Comment:
@Asmaa-ezz I think you could use `queryString.stringify()` to put all of these query parameters together for you, if you'd like:
```
url = `api/v1/filter?${ queryString.stringify( values ) };`
```
Here's the docs:
https://github.com/sindresorhus/query-string#stringifyobject-options